proposals/0004.md

```diff
@@ -0,0 +1,153 @@
+* Title: *Radicle client URI scheme*
+* Champion: *@rudolfs, rudolfs@osins.org*
+* Conversation:
+  * [Radicle Community discussion][community]
+  * [Shareable identifier discussion][shareable]
+  * [Deep linking discussion][deeplink]
+* References:
+  * [List of URI schemes][urilist]
+
+# Motivation
+
+We have received multiple feature requests from our community regarding
+deep-linking in our client application. A common way to enable such
+functionality is to devise a URI scheme along with a custom protocol and
+implement a handler which listens for requests on the given protocol in a
+client application.
+
+Given that there might be multiple competing Radicle client implementations and
+to make sure that shared URIs are compatible with as many Radicle client
+implementations as possible we decided to have a specification for our URI
+scheme.
+
+
+# Explanation
+
+The main use case for a custom protocol is to be able to open a Radicle client
+application by clicking on a link on the web in an external browser. The
+Radicle client should be able to recognise what kind of resource is being
+requested and be able to render the request accordingly.
+
+Besides showing resources it might be valuable to perform certain actions on
+said resources. For example a user may be able to start tracking a project by
+clicking a link or initiate a fund transfer to a certain address.
+
+
+# Recommendation
+
+To make implementation easier we want to use existing URI parsing libraries,
+therefore our URI scheme should be parsable according to [RFC 3986][rfc3986].
+
+To make sure our URI scheme is extensible in the future and clients can provide
+backwards compatibility for older links we should have a version identifier in
+the scheme.
+
+The URI scheme to be used by Radicle client applications should adhere to the
+following scheme:
+
+    radicle://[<protocolVersion>/<context>/<entity>?action=<action>]
+
+_Note:_ brackets denote optionality.
+
+Where:
+
+- `radicle:` is the Radicle client protocol name.
+- `<protocolVersion>` is a version number of the Radicle client protocol. This
```

Comment:
How do we expect the protocol version to interplay with link's protocol version? I'd imagine it would need to be bumped if, for example, the URN parser changed again for some reason.

Reply:
Good point. I think it makes sense to move the `<protocolVersion>` after the `<context>` as @kim [suggested](https://github.com/radicle-dev/radicle-decisions/pull/10#discussion_r592978118). This way the version will be scoped to the context, and whenever there's a change to the `link` context we'll bump its version.

Reply:
Have a look at the revised spec. Does the current form do the job?